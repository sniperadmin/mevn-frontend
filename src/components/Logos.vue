<template>
  <v-row justify="center">
    <v-col cols="2" sm="4">
      <v-img alt="Mongo logo"
        src="../assets/mongodb.png"
        contain
        class="shrink mt-3"
        height="200"
      />
    </v-col>
    <v-col cols="2" sm="4">
      <v-img alt="Express logo"
        src="../assets/express-min.png"
        contain
        class="shrink mt-3"
        height="200"
      />
    </v-col>
    <v-col cols="2" sm="4">
      <v-img
        src="../assets/logo.svg"
        contain
        height="200"
      />
    </v-col>
    <v-col cols="2" sm="4">
      <v-img alt="Vue logo"
        src="../assets/logo.png"
        contain
        class="shrink mt-3"
        height="200"
      />
    </v-col>
    <v-col cols="2" sm="4">
      <v-img alt="Nodejs logo"
        src="../assets/nodejs.png"
        contain
        class="shrink mt-3"
        height="200"
      />
    </v-col>
    <v-col cols="8" sm="4">
      <v-img alt="Nodejs logo"
        src="../assets/jest.png"
        contain
        class="shrink mt-3"
        height="200"
      />
    </v-col>
  </v-row>
</template>

<script>
  export default {
    name: 'Logos',
    data: () => ({})
  }
</script>

<style scoped>

</style>