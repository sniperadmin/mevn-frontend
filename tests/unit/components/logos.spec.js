import Vue from 'vue'
import { mount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'
import Logos from '@/components/Logos.vue'

describe('Logos', () => {
  let wrapper
  const localVue = createLocalVue()

  Vue.use(Vuetify)

  const vuetify = new Vuetify()

  beforeEach(() => {
    wrapper = mount(Logos, {
      localVue,
      vuetify
    })
  })

  afterEach(() => {
    wrapper.destroy()
  })
  
  it('should load images', () => {
    const logos = wrapper.findAll('.v-image__image')

    expect(wrapper.html()).toMatchSnapshot()
    expect(logos.length).toEqual(6)
  })
})