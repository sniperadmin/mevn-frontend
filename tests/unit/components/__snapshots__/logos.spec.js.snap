// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logos should load images 1`] = `
<div class="row justify-center">
  <div class="col-sm-4 col-2">
    <div role="img" aria-label="Mongo logo" class="v-responsive v-image shrink mt-3" style="height: 200px;">
      <transition-stub name="fade-transition" mode="in-out">
        <div class="v-image__image v-image__image--preload v-image__image--contain" style="background-position: center center;"></div>
      </transition-stub>
      <div class="v-responsive__content"></div>
    </div>
  </div>
  <div class="col-sm-4 col-2">
    <div role="img" aria-label="Express logo" class="v-responsive v-image shrink mt-3" style="height: 200px;">
      <transition-stub name="fade-transition" mode="in-out">
        <div class="v-image__image v-image__image--preload v-image__image--contain" style="background-position: center center;"></div>
      </transition-stub>
      <div class="v-responsive__content"></div>
    </div>
  </div>
  <div class="col-sm-4 col-2">
    <div class="v-responsive v-image" style="height: 200px;">
      <transition-stub name="fade-transition" mode="in-out">
        <div class="v-image__image v-image__image--preload v-image__image--contain" style="background-position: center center;"></div>
      </transition-stub>
      <div class="v-responsive__content"></div>
    </div>
  </div>
  <div class="col-sm-4 col-2">
    <div role="img" aria-label="Vue logo" class="v-responsive v-image shrink mt-3" style="height: 200px;">
      <transition-stub name="fade-transition" mode="in-out">
        <div class="v-image__image v-image__image--preload v-image__image--contain" style="background-position: center center;"></div>
      </transition-stub>
      <div class="v-responsive__content"></div>
    </div>
  </div>
  <div class="col-sm-4 col-2">
    <div role="img" aria-label="Nodejs logo" class="v-responsive v-image shrink mt-3" style="height: 200px;">
      <transition-stub name="fade-transition" mode="in-out">
        <div class="v-image__image v-image__image--preload v-image__image--contain" style="background-position: center center;"></div>
      </transition-stub>
      <div class="v-responsive__content"></div>
    </div>
  </div>
  <div class="col-sm-4 col-8">
    <div role="img" aria-label="Nodejs logo" class="v-responsive v-image shrink mt-3" style="height: 200px;">
      <transition-stub name="fade-transition" mode="in-out">
        <div class="v-image__image v-image__image--preload v-image__image--contain" style="background-position: center center;"></div>
      </transition-stub>
      <div class="v-responsive__content"></div>
    </div>
  </div>
</div>
`;
